plugins/image-prioritizer/helper.php

```diff
@@ -235,6 +235,15 @@ static function ( $host ) {
 	// Validate that the Content-Type is an image.
 	$content_type = (array) wp_remote_retrieve_header( $r, 'content-type' );
 	if ( ! is_string( $content_type[0] ) || ! str_starts_with( $content_type[0], 'image/' ) ) {
+		// This is a special case for images that are not served with the correct Content-Type.
+		if ( is_string( $content_type[0] ) && 'application/octet-stream' === $content_type[0] ) {
+			$file_extension              = pathinfo( (string) wp_parse_url( $url, PHP_URL_PATH ), PATHINFO_EXTENSION );
+			$valid_image_file_extensions = array( 'jpg', 'jpeg', 'png', 'gif', 'webp', 'avif', 'heic', 'heif', 'svg' );
+			if ( '' !== $file_extension && in_array( strtolower( $file_extension ), $valid_image_file_extensions, true ) ) {
+				return true;
```

Comment:
This doesn't look right because then the `$content_length` check below is not being accounted for.